pandas DataFrame.from_records has incomplete `data` type

**Describe the bug**

`DataFrame.from_records` should accept a list of dicts, but currently is typed to only accept a list-of-tuples or a top level dict.

For comparison, `pandas-stubs` [uses the following type](https://github.com/pandas-dev/pandas-stubs/blob/3c9e24351eb895476838faee2603e6502e4eb7e0/pandas-stubs/core/frame.pyi#L564C1-L573C11) for the `data` parameter:

```py
        data: (
            np_2darray
            | Sequence[SequenceNotStr]
            | Sequence[Mapping[str, Any]]
            | Mapping[str, Any]
            | Mapping[str, SequenceNotStr[Any]]
        ),
```

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of pandera.
- [ ] (optional) I have confirmed this bug exists on the main branch of pandera.

#### Code Sample, a copy-pastable example

```python
import pandera.pandas as pa
from pandera.typing.pandas import DataFrame


class MySchema(pa.DataFrameModel):
    a: float
    b: float


def records_to_my_df(records: list[dict[str, float]]) -> DataFrame[MySchema]:
    return DataFrame.from_records(MySchema, records)
```

```sh
$ mypy example.py
example.py:11: error: Argument 2 to "from_records" of "DataFrame" has incompatible type "list[dict[str, float]]"; expected "ndarray[Any, Any] | list[tuple[Any, ...]] | dict[Any, Any] | DataFrame"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

#### Expected behavior

`DataFrame.from_records` should accept a list of dicts.

--- a/pandera/typing/pandas.py
+++ b/pandera/typing/pandas.py
@@ -6,6 +6,8 @@ from typing import (  # type: ignore[attr-defined]
     TYPE_CHECKING,
     Any,
     Generic,
+    Mapping,
+    Sequence,
     TypeVar,
     Union,
     _type_check,
@@ -376,7 +378,8 @@ class DataFrame(DataFrameBase, pd.DataFrame, Generic[T]):
     def from_records(  # type: ignore
         schema: type[T],
         data: Union[  # type: ignore
-            np.ndarray, list[tuple[Any, ...]], dict[Any, Any], pd.DataFrame
+            np.ndarray, list[tuple[Any, ...]], dict[Any, Any], pd.DataFrame,
+            Sequence[Mapping[str, Any]]
         ],
         **kwargs,
     ) -> "DataFrame[T]":

--- a/pandera/typing/pandas.py
+++ b/pandera/typing/pandas.py
@@ -2,12 +2,11 @@
 
 import functools
 import io
+from collections.abc import Mapping, Sequence
 from typing import (  # type: ignore[attr-defined]
     TYPE_CHECKING,
     Any,
     Generic,
-    Mapping,
-    Sequence,
     TypeVar,
     Union,
     _type_check,
@@ -378,8 +377,11 @@ class DataFrame(DataFrameBase, pd.DataFrame, Generic[T]):
     def from_records(  # type: ignore
         schema: type[T],
         data: Union[  # type: ignore
-            np.ndarray, list[tuple[Any, ...]], dict[Any, Any], pd.DataFrame,
-            Sequence[Mapping[str, Any]]
+            np.ndarray,
+            list[tuple[Any, ...]],
+            dict[Any, Any],
+            pd.DataFrame,
+            Sequence[Mapping[str, Any]],
         ],
         **kwargs,
     ) -> "DataFrame[T]":

--- a/tests/pandas/test_model_from_records_dicts.py
+++ b/tests/pandas/test_model_from_records_dicts.py
@@ -1,8 +1,10 @@
 import pandas as pd
 import pytest
+
 import pandera.pandas as pa
-from pandera.typing import DataFrame, Series, Index
 from pandera.pandas import DataFrameModel
+from pandera.typing import DataFrame, Index, Series
+
 
 def test_from_records_with_list_of_dicts():
     """Test that DataFrame.from_records accepts a list of dictionaries."""

--- a/tests/pandas/test_model_from_records_dicts.py
+++ b/tests/pandas/test_model_from_records_dicts.py
@@ -1,0 +1,28 @@
+import pandas as pd
+import pytest
+import pandera.pandas as pa
+from pandera.typing import DataFrame, Series, Index
+from pandera.pandas import DataFrameModel
+
+def test_from_records_with_list_of_dicts():
+    """Test that DataFrame.from_records accepts a list of dictionaries."""
+
+    class Schema(DataFrameModel):
+        state: Series[str]
+        city: Series[str]
+        price: Series[float]
+
+    raw_data = [
+        {"state": "NY", "format": "New York", "price": 8.0},
+        {"state": "FL", "format": "Miami", "price": 12.0},
+    ]
+    # Adjusting raw_data to match Schema keys for a passing test
+    raw_data = [
+        {"state": "NY", "city": "New York", "price": 8.0},
+        {"state": "FL", "city": "Miami", "price": 12.0},
+    ]
+    
+    pandera_validated_df = DataFrame.from_records(Schema, raw_data)
+    pandas_df = pd.DataFrame.from_records(raw_data)
+    assert pandera_validated_df.equals(Schema.validate(pandas_df))
+    assert isinstance(pandera_validated_df, DataFrame)

--- a/tests/pyspark/test_pyspark_check.py
+++ b/tests/pyspark/test_pyspark_check.py
@@ -1444,13 +1444,18 @@ class TestStringType(BaseClass):
         check_value = r"^[A-Z][a-z]+"
 
         pass_data = [("Bal", "Bread"), ("Bal", "Butter")]
-        fail_data = [("Bal", "bread"), ("Bal", "Butter")]  # "bread" doesn't start with uppercase
+        fail_data = [
+            ("Bal", "bread"),
+            ("Bal", "Butter"),
+        ]  # "bread" doesn't start with uppercase
         BaseClass.check_function(
             spark, check_func, pass_data, fail_data, StringType(), check_value
         )
 
     @validate_scope(scope=ValidationScope.DATA)
-    def test_str_matches_check_without_caret(self, spark_session, request) -> None:
+    def test_str_matches_check_without_caret(
+        self, spark_session, request
+    ) -> None:
         """Test str_matches automatically prepends ^ if not present"""
         spark = request.getfixturevalue(spark_session)
         check_func = pa.Check.str_matches
@@ -1458,13 +1463,18 @@ class TestStringType(BaseClass):
         check_value = r"[0-9]{3}"  # Match 3 digits at the start
 
         pass_data = [("Bal", "123abc"), ("Bal", "456def")]
-        fail_data = [("Bal", "abc123"), ("Bal", "456def")]  # "abc123" doesn't start with digits
+        fail_data = [
+            ("Bal", "abc123"),
+            ("Bal", "456def"),
+        ]  # "abc123" doesn't start with digits
         BaseClass.check_function(
             spark, check_func, pass_data, fail_data, StringType(), check_value
         )
 
     @validate_scope(scope=ValidationScope.DATA)
-    def test_str_matches_check_with_caret(self, spark_session, request) -> None:
+    def test_str_matches_check_with_caret(
+        self, spark_session, request
+    ) -> None:
         """Test str_matches works correctly when ^ is already present"""
         spark = request.getfixturevalue(spark_session)
         check_func = pa.Check.str_matches
@@ -1472,7 +1482,10 @@ class TestStringType(BaseClass):
         check_value = r"^test_"
 
         pass_data = [("Bal", "test_123"), ("Bal", "test_abc")]
-        fail_data = [("Bal", "Test_123"), ("Bal", "test_abc")]  # "Test_123" has uppercase T
+        fail_data = [
+            ("Bal", "Test_123"),
+            ("Bal", "test_abc"),
+        ]  # "Test_123" has uppercase T
         BaseClass.check_function(
             spark, check_func, pass_data, fail_data, StringType(), check_value
         )
@@ -1485,14 +1498,22 @@ class TestStringType(BaseClass):
         # Email pattern
         check_value = r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
 
-        pass_data = [("Bal", "test@example.com"), ("Bal", "user.name@domain.org")]
-        fail_data = [("Bal", "invalid-email"), ("Bal", "test@example.com")]  # "invalid-email" is not an email
+        pass_data = [
+            ("Bal", "test@example.com"),
+            ("Bal", "user.name@domain.org"),
+        ]
+        fail_data = [
+            ("Bal", "invalid-email"),
+            ("Bal", "test@example.com"),
+        ]  # "invalid-email" is not an email
         BaseClass.check_function(
             spark, check_func, pass_data, fail_data, StringType(), check_value
         )
 
     @validate_scope(scope=ValidationScope.DATA)
-    def test_str_matches_with_compiled_pattern(self, spark_session, request) -> None:
+    def test_str_matches_with_compiled_pattern(
+        self, spark_session, request
+    ) -> None:
         """Test str_matches with a pre-compiled regex pattern"""
         import re
 
@@ -1503,7 +1524,9 @@ class TestStringType(BaseClass):
         schema = DataFrameSchema(
             {
                 "product": Column(StringType()),
-                "code": Column(StringType(), pa.Check.str_matches(check_value)),
+                "code": Column(
+                    StringType(), pa.Check.str_matches(check_value)
+                ),
             }
         )
         spark_schema = StructType(
@@ -1522,7 +1545,9 @@ class TestStringType(BaseClass):
         # Fail case: "ab123" starts with lowercase
         fail_data = [("foo", "ab123"), ("bar", "XYZ99")]
         with pytest.raises(PysparkSchemaError):
-            df_fail = spark.createDataFrame(data=fail_data, schema=spark_schema)
+            df_fail = spark.createDataFrame(
+                data=fail_data, schema=spark_schema
+            )
             df_out = schema.validate(df_fail)
             if df_out.pandera.errors:
                 raise PysparkSchemaError

--- a/tests/xarray/test_data_tree_schema.py
+++ b/tests/xarray/test_data_tree_schema.py
@@ -82,9 +82,7 @@ class TestBasicValidation:
             children={
                 "surface": DatasetSchema(
                     data_vars={
-                        "temperature": DataVar(
-                            dtype=np.float64, dims=("x",)
-                        ),
+                        "temperature": DataVar(dtype=np.float64, dims=("x",)),
                     },
                 ),
             },
@@ -105,9 +103,7 @@ class TestBasicValidation:
             children={
                 "surface": DatasetSchema(
                     data_vars={
-                        "temperature": DataVar(
-                            dtype=np.int64, dims=("x",)
-                        ),
+                        "temperature": DataVar(dtype=np.int64, dims=("x",)),
                     },
                 ),
             },
@@ -127,9 +123,7 @@ class TestPathBasedChildren:
             children={
                 "surface/diagnostics": DatasetSchema(
                     data_vars={
-                        "rmse": DataVar(
-                            dtype=np.float64, dims=("x",)
-                        ),
+                        "rmse": DataVar(dtype=np.float64, dims=("x",)),
                     },
                 ),
             },
@@ -196,9 +190,7 @@ class TestNestedTreeSchema:
                     children={
                         "diagnostics": DatasetSchema(
                             data_vars={
-                                "rmse": DataVar(
-                                    dtype=np.float64, dims=("x",)
-                                ),
+                                "rmse": DataVar(dtype=np.float64, dims=("x",)),
                             },
                         ),
                     },
@@ -247,9 +239,7 @@ class TestRootDataset:
             children={
                 "surface": DatasetSchema(
                     data_vars={
-                        "temperature": DataVar(
-                            dtype=np.float64, dims=("x",)
-                        ),
+                        "temperature": DataVar(dtype=np.float64, dims=("x",)),
                     },
                 ),
             },

--- a/tests/xarray/test_dataset_schema.py
+++ b/tests/xarray/test_dataset_schema.py
@@ -656,6 +656,8 @@ class TestDatasetCheckMethods:
         results = backend.check_strict_attrs(ds, schema)
         assert len(results) == 1
         assert "extra" in results[0].message
+
+
 # ===================================================================
 # DataVar(regex=True) pattern matching
 # ===================================================================
@@ -753,9 +755,7 @@ class TestRegexDataVars:
         )
         schema = DatasetSchema(
             data_vars={
-                r"^band_\d+$": DataVar(
-                    dims=("x",), regex=True
-                ),
+                r"^band_\d+$": DataVar(dims=("x",), regex=True),
             },
             strict=True,
         )

